addons/test_website_modules/static/tests/tours/configurator_flow.js

```diff
@@ -88,7 +88,7 @@ registry.category("web_tour.tours").add('configurator_flow', {
                 }
             }
             for (const url of ['/', '/event', '/slides', '/pricing', '/blog/', '/blog/', '/contactus']) {
-                if (!$iframe.contents().find(`#top_menu a[href^='${url}'`).length) {
+                if (!$iframe.contents().find(`#top_menu a[href^='${url}']`).length) {
```

Comment:
@detrouxdev I am not sure who wrote this test as it's a multi author commit including you but maybe you know: I have the impression that this `run()` is actually never tested, is it?
From what I understand, won't the testing engine just find the `trigger` here (which succeed) and then try to find the next trigger (and at the same time start the run code) but since there is none, it will just mark the test as succeed, actually before the run is even executed?
Generally, when we check something in a custom `run()`, at the end we add a class or something so we can check its presence in the next step (to be sure we went through the full run and actually indirectly wait for it to be executed).

A bit like the `assertCssVariable` which IMO seems to be wrong as it does a check in the run but the next step is not waiting for it, so it will crash correctly because there is many steps after it taking some time which will leave time for the JS to actually crash, by chance?

tl;dr: The content of a `run()` method is not awaited to proceed to the next step. The next step's trigger of a tour should thus never be found before running the current step, which is the case here: nothing is ensuring that this step operation is done properly.

I wrote a wip back in the days to ensure the next step is not reachable but it seems to be hard to make automatic
https://github.com/odoo-dev/odoo/commit/2ef6ca12894ff46fe6c32c13fe39f866c95a70fd

Reply:
Note that
```js
$iframe.contents().find(`#top_menu a[href^='${url}']`
```
Is not crashing from what we tested with ARD, jquery is magical enough to fix the selector for you somehow and find the element.

We will check to be sure what I said is right: run methods are not awaited and next step will ignore it / not wait before next step

![image](https://github.com/odoo/odoo/assets/30048408/739582fa-092a-48e3-9633-6b6413430205)

Reply:
I actually understand what's going on, i'm on chrome 113 locally 

This is actually also breaking on runbot in the latest chrome build 
https://runbot.odoo.com/runbot/build/36161837